forge-gui-desktop/src/main/java/forge/screens/home/VLobby.java

```diff
@@ -353,14 +298,23 @@ public void update(final boolean fullUpdate) {
                 panel.update();
 
                 final boolean isSlotAI = slot.getType() == LobbySlotType.AI;
-
-                deckChooser.setIsAi(isSlotAI);
-                commanderDeckChooser.setIsAi(isSlotAI);
-                oathbreakerDeckChooser.setIsAi(isSlotAI);
-                tinyLeaderDeckChooser.setIsAi(isSlotAI);
-                brawlDeckChooser.setIsAi(isSlotAI);
+                if (isNewPanel || fullUpdate) {
+                    final FDeckChooser deckChooser = createDeckChooser(lobby.getGameType(), i, isSlotAI);
+                    deckChooser.populate();
+                    panel.setDeckChooser(deckChooser);
+                    if (i == 0) {
+                        // TODO: This seems like the wrong place to do this:
+                        slot.setIsDevMode(prefs.getPrefBoolean(FPref.DEV_MODE_ENABLED));
+                        changePlayerFocus(0);
+                    }
+                } else {
+                    panel.getDeckChooser().setIsAi(isSlotAI);
+                }
                 if (fullUpdate && (type == LobbySlotType.LOCAL || isSlotAI)) {
-                    selectDeck(i);
+                    // Deck section selection
```

Comment:
Ok, so your change here breaks this: 72d40d7f49c52df727b627a02974a59a1a26f51d

I feel like there could be situations where either behaviour is desired though:
1. wanting a redemption match with the same decks
2. expecting the same kind of challenge again where each player has to deal with new decks

As it is the GUI doesn't really make it clear more interaction is needed to reach state 2 instead of 1 now...

Also if we change the default I'm not really sure if this Lobby instance gets reused for some other things like the command line match simulations which expect to be provided new decks?

I would recommend a new checkbox in the lobby that can be enabled to skip the refresh

But if you have some other idea feel free to go with that :)

Reply:
Thanks! That wasn't intentional, so I've restored the previous behavior.

I agree that it's debatable which is better, but I can see the argument for the existing behavior being more desirable, as the case where you want a rematch can already be achieved from the dialog when the match ends, you can do a re-match.

So I've restored to how it was before. Please take another look.